### PR TITLE
DSND-2353: Remove set mapping in transform rules with missing capture groups

### PR DESCRIPTION
**Pull request by johncook-ch**
## Describe the changes
* Edited 2 rules to remove set clauses where the `chargeCreationDate` capture group is never populated by a like regex
* Added an example delta and expected PUT body to ConsumerPositiveComprehensiveIT
* Added the transaction ID when logging the chosen transformation rule

### Related Jira tickets
[DSND-2353](https://companieshouse.atlassian.net/browse/DSND-2353)

### General
- [x] Is the PR as small as it can be?
- [x] Has the Jira ticket been updated with any relevant comments?
- [x] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [x] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2353]: https://companieshouse.atlassian.net/browse/DSND-2353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sthompsonCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/resources/transform_rules.yml [hunk 1]
```diff
@@ -4751,8 +4747,6 @@
       data.description_values.charge_number: '[% chargeNumber %]'
       data.subcategory: alter
       original_description: '[% data.description | sentence_case %]'
-      data.action_date: '[% chargeCreationDate | bson_date %]'
-      data.description_values.charge_creation_date: '[% chargeCreationDate | bson_date %]'
```
Comment (cgregoryCH) [inline src/main/resources/transform_rules.yml:4755]: This doesn't look like a fall back rule, rather one which is missing an appropriate like clause. The charge_creation_date is expected for this description. https://github.com/companieshouse/api-enumerations/blob/d930c7b15828f98f1ac2f28ac704ef130fd602b1/filing_history_descriptions.yml#L463

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/resources/transform_rules.yml (hunk 1 above)
Comment (johncook-ch) [reply src/main/resources/transform_rules.yml]: Restored the rule and created a new bug to cover the fix for when substitution values are in the delta's `description_values` field and not drawn from the capture groups
https://companieshouse.atlassian.net/browse/DSND-2534

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/transformrules/TransformerService.java [hunk 2]
```diff
@@ -27,7 +27,8 @@ public JsonNode transform(JsonNode delta) {
             Result result = rule.match(delta);
             if (result.matched()) {
                 When when = rule.when();
-                LOGGER.info("Matched transform rule: [eq: %s, like: %s]".formatted(when.formType(), when.like()));
+                LOGGER.info("Transaction %s, matched transform rule: [eq: %s, like: %s]"
+                        .formatted(delta.get("_entity_id").textValue(), when.formType(), when.like()));
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/transformrules/TransformerService.java:31]: Missing the DataMapHolder logMap

**Review by cgregoryCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**